ocrd/ocrd/decorators.py

```diff
@@ -46,9 +56,25 @@ def ocrd_cli_wrap_processor(processorClass, ocrd_tool=None, mets=None, working_d
             raise Exception(msg)
         resolver = Resolver()
         workspace = resolver.workspace_from_url(mets, working_dir)
-        # TODO once we implement 'overwrite' CLI option and mechanism, disable the
-        # `output_file_grp_ check by setting to False-y value if 'overwrite' is set
-        report = WorkspaceValidator.check_file_grp(workspace, kwargs['input_file_grp'], kwargs['output_file_grp'])
+        if overwrite:
+            if 'output_file_grp' not in kwargs or not kwargs['output_file_grp']:
+                raise Exception("--overwrite requires --output-file-grp")
+            LOG.debug("Removing files because of --overwrite")
+            for grp in kwargs['output_file_grp'].split(','):
+                if 'page_id' in kwargs:
+                    for page_id in kwargs['page_id'].split(','):
+                        LOG.debug("Removing files in output file group %s with page ID %s", grp, page_id)
+                        for file in workspace.mets.find_files(pageId=page_id, fileGrp=grp):
+                            workspace.remove_file(file, force=True, keep_file=False)
+                else:
+                    LOG.debug("Removing all files in output file group %s ", grp)
+                    workspace.remove_file_group(grp, recursive=True, force=True, keep_files=False)
```

Comment:
As I have warned before, this is – unfortunately – not enough. PAGE files can reference derived image files in their respective file groups. You _must_ identify these. (It's probably worth doing this only once in a re-usable function. There's similar demand for derived image dependency tracking in other contexts.)

Reply:
Good point, I missed that completely. Can you pls provide a small-ish workspace with rich use of AlternativeImage for addition to assets so I can test? Thanks

Reply:
But how do I share it with you?

I recommend just running `gt-binarize-page-olena-sauvola-denoise-ocropy-deskew-page-ocropy-clip-deskew-region-tesseract-resegment-dewarp-ocr-ocropy-tesseract-extract-lines.mk` on `kant_aufklaerung` yourself.

Reply:
I've added a PAGE API method `get_AllImagePaths` to get all image paths in a PAGE-XML document.

`Workspace.remove_file` now has two additional parameters `page_recursive` and `page_same_group`. If the former is `True` (the default), then on removal of a PAGE-XML file, it is also parsed for paths of `pc:Page/@imageFilename` and `pc:AlternativeImage/@filename` and those are recursively removed as well.

The idea behind the `page_same_region` parameter is to restrict deleting images to those that are in the same `pc:fileGrp` as the PAGE-XML document is, e.g. keeping the image referenced in `@imageFilename` if the PAGE-XML document is derived from another PAGE-XML document. However, I cannot test this, since I don't have any test data that follows this convention we agreed on in the last Tech Call (that AlternativeImages should be written to the fileGrp of the PAGE-XML that references them).

Also I'm interested in PAGE-XML samples with AlternativeImage on Word and Glyph level. If anybody can provide, I'd appreciate it!

Reply:
> The idea behind the `page_same_group` parameter is to restrict deleting images to those that are in the same `pc:fileGrp` as the PAGE-XML document is, e.g. keeping the image referenced in `@imageFilename` if the PAGE-XML document is derived from another PAGE-XML document. However, I cannot test this, since I don't have any test data that follows this convention we agreed on in the last Tech Call (that AlternativeImages should be written to the fileGrp of the PAGE-XML that references them).

Yes, this is a hen-or-egg problem. I think we could either start a feature branch in core and at least one preprocessing module to test this realistically, or add a pseudo-processor in the test section here (perhaps derived from the new dummy processor) which produces a dummy AlternativeImage via `image_from_segment` and subsequent `save_image_file`.

Reply:
> Also I'm interested in PAGE-XML samples with AlternativeImage on Word and Glyph level. If anybody can provide, I'd appreciate it!

Sorry, I don't know of any processor that creates derived images below the line level.

Reply:
Tracking in https://github.com/OCR-D/core/issues/505